parser/gotest/report_builder.go

```diff
@@ -278,11 +278,18 @@ func (b *reportBuilder) findTest(name string) (int, bool) {
 	if t, ok := b.tests[b.lastID]; ok && t.Name == name {
 		return b.lastID, true
 	}
+	firstMatch := -1
 	for i := b.nextID; i > 0; i-- {
 		if test, ok := b.tests[i]; ok && test.Name == name {
-			return i, true
+			firstMatch = i
+			if test.Result == gtr.Unknown {
+				return i, true
+			}
```

Comment:
should this change not continue to look further till `test.Result != gtr.Unknown`

Reply:
No, we want to use results with `gtr.Unknown` as they will be set with their proper value right after the call to `findTest`.